src/pages/explore/api/get-stats.ts

```diff
@@ -0,0 +1,97 @@
+import { DurationWindow } from '@/shared/utils/duration';
+import { toValueView } from '@/shared/utils/value-view';
+
+export interface StatsData {
+  activePairs: number;
+  trades: number;
+  largestPair?: { start: string; end: string };
+  topPriceMover?: { start: string; end: string; percent: number };
+  directVolume: ValueView;
+  liquidity: ValueView;
+  largestPairLiquidity?: ValueView;
+}
+
+export type StatsResponse = StatsData | { error: string };
+
+export const getStats = async (): Promise<StatsResponse> => {
+  try {
+    const durationWindow: DurationWindow = '1d';
+
+    const chainId = process.env['PENUMBRA_CHAIN_ID'];
+    if (!chainId) {
+      return { error: 'PENUMBRA_CHAIN_ID is not set' };
+    }
+
+    const registryClient = new ChainRegistryClient();
+
+    const [registry, results] = await Promise.all([
+      registryClient.remote.get(chainId),
+      pindexer.stats(durationWindow),
+    ]);
+
+    const stats = results[0];
+    if (!stats) {
+      return { error: `No stats found` };
+    }
+
+    // TODO: Add getMetadataBySymbol() helper to registry npm package
+    const allAssets = registry.getAllAssets();
+    // TODO: what asset should be used here?
+    const usdcMetadata = allAssets.find(asset => asset.symbol.toLowerCase() === 'usdc');
+    if (!usdcMetadata) {
+      return { error: 'USDC not found in registry' };
+    }
+
+    const topPriceMoverStart = allAssets.find(asset => {
+      return asset.penumbraAssetId?.equals(new AssetId({ inner: stats.top_price_mover_start }));
+    });
+    const topPriceMoverEnd = allAssets.find(asset => {
+      return asset.penumbraAssetId?.equals(new AssetId({ inner: stats.top_price_mover_end }));
+    });
+    const topPriceMover = topPriceMoverStart &&
+      topPriceMoverEnd && {
+        start: topPriceMoverStart.symbol,
+        end: topPriceMoverEnd.symbol,
+        percent: stats.top_price_mover_change_percent,
+      };
+
+    const largestPairStart = allAssets.find(asset => {
+      return asset.penumbraAssetId?.equals(
+        new AssetId({ inner: stats.largest_dv_trading_pair_start }),
+      );
+    });
+    const largestPairEnd = allAssets.find(asset => {
+      return asset.penumbraAssetId?.equals(
+        new AssetId({ inner: stats.largest_dv_trading_pair_end }),
+      );
+    });
+    const largestPair = largestPairStart &&
+      largestPairEnd && {
+        start: largestPairStart.symbol,
+        end: largestPairEnd.symbol,
+      };
+
+    return {
+      activePairs: stats.active_pairs,
+      trades: stats.trades,
+      largestPair,
+      topPriceMover,
+      largestPairLiquidity:
+        largestPairEnd &&
+        toValueView({
+          amount: stats.largest_dv_trading_pair_volume,
+          metadata: largestPairEnd,
+        }),
+      liquidity: toValueView({
+        amount: parseInt(`${stats.liquidity}`),
+        metadata: usdcMetadata,
+      }),
+      directVolume: toValueView({ amount: stats.direct_volume, metadata: usdcMetadata }),
```

Comment:
but it doesn't send this over the wire since everything is rendered in react server components with `use server` (or without `use client`, to be precise). The data is fetched and immediately transformed to HTML. This valid HTML is then sent over the wire to the client.